src/ArraySizeException.java

```diff
@@ -0,0 +1,5 @@
+public class ArraySizeException extends Exception {
```

Comment:
Extending from `Exception` is not really correct. Would be enough to extend from `RuntimeException`